deploy/005_deploy_arbitrum_spokepool.ts

```diff
@@ -16,17 +13,30 @@ const func = async function (hre: HardhatRuntimeEnvironment) {
 
   const chainId = parseInt(await getChainId());
 
-  await deploy("Arbitrum_SpokePool", {
-    from: deployer,
-    log: true,
-    skipIfAlreadyDeployed: true,
-    args: [
-      L2_ADDRESS_MAP[chainId].l2GatewayRouter, // _l2GatewayRouter
-      hubPool.address, // Set hub pool as cross domain admin since it delegatecalls the Optimism_Adapter logic.
-      hubPool.address,
-      L2_ADDRESS_MAP[chainId].l2Weth, // l2Weth
-      "0x0000000000000000000000000000000000000000", // timer
-    ],
+  // Initialize deposit counter to very high number of deposits to avoid duplicate deposit ID's
+  // with deprecated spoke pool.
+  // Set hub pool as cross domain admin since it delegatecalls the Adapter logic.
+  const constructorArgs = [
+    1_000_000,
+    L2_ADDRESS_MAP[chainId].l2GatewayRouter,
+    hubPool.address,
+    hubPool.address,
+    L2_ADDRESS_MAP[chainId].l2Weth,
+  ];
+  const spokePool = await upgrades.deployProxy(await getContractFactory("Arbitrum_SpokePool"), constructorArgs, {
+    kind: "uups",
+  });
+  const instance = await spokePool.deployed();
+  console.log(`SpokePool deployed @ ${instance.address}`);
+  const implementationAddress = await upgrades.erc1967.getImplementationAddress(instance.address);
+  console.log(`Implementation deployed @ ${implementationAddress}`);
+
+  // hardhat-upgrades overrides the `verify` task that ships with `hardhat` so that if the address passed
+  // is a proxy, hardhat will first verify the implementation and then the proxy and also link the proxy
+  // to the implementation's ABI on etherscan.
+  // https://docs.openzeppelin.com/upgrades-plugins/1.x/api-hardhat-upgrades#verify
+  await run("verify:verify", {
+    address: instance.address,
```

Comment:
This part seems pretty repetitive across the various deployment scripts - might be ripe for a utility function that handles all this. wdyt?

Reply:
yeah thats reasonable